code/game/objects/items/storage/toolbox.dm

```diff
@@ -11,75 +11,61 @@
 	w_class = WEIGHT_CLASS_BULKY
 	attack_verb = list("robusted")
 
-
 /obj/item/storage/toolbox/emergency
 	name = "emergency toolbox"
 	icon_state = "red"
 	item_state = "toolbox_red"
-
-
-/obj/item/storage/toolbox/emergency/Initialize(mapload, ...)
-	. = ..()
-	new /obj/item/tool/crowbar/red(src)
-	new /obj/item/tool/extinguisher/mini(src)
-	if(prob(50))
-		new /obj/item/flashlight(src)
-	else
-		new /obj/item/flashlight/flare(src)
-	new /obj/item/radio(src)
-
+	spawns_with = list(
+		/obj/item/tool/crowbar/red,
+		/obj/item/tool/extinguisher/mini,
+		/obj/item/flashlight/flare,
+		/obj/item/radio,
+	)
 
 /obj/item/storage/toolbox/mechanical
 	name = "mechanical toolbox"
 	icon_state = "blue"
 	item_state = "toolbox_blue"
-
-
-/obj/item/storage/toolbox/mechanical/Initialize(mapload, ...)
-	. = ..()
-	new /obj/item/tool/screwdriver(src)
-	new /obj/item/tool/wrench(src)
-	new /obj/item/tool/weldingtool(src)
-	new /obj/item/tool/crowbar(src)
-	new /obj/item/analyzer(src)
-	new /obj/item/tool/wirecutters(src)
-
+	spawns_with = list(
+		/obj/item/tool/screwdriver,
+		/obj/item/tool/wrench,
+		/obj/item/tool/weldingtool,
+		/obj/item/tool/crowbar,
+		/obj/item/analyzer,
+		/obj/item/tool/wirecutters,
+	)
 
 /obj/item/storage/toolbox/electrical
 	name = "electrical toolbox"
 	icon_state = "yellow"
 	item_state = "toolbox_yellow"
-
-
-/obj/item/storage/toolbox/electrical/Initialize(mapload, ...)
+	spawns_with = list(
+		/obj/item/tool/screwdriver,
+		/obj/item/tool/wirecutters,
+		/obj/item/t_scanner,
+		/obj/item/tool/crowbar,
+	)
+	spawns_prob = list(list(5, /obj/item/clothing/gloves/yellow))
+
+/obj/item/storage/toolbox/electrical/PopulateContents()
 	. = ..()
-	var/color = pick("red", "yellow", "green", "blue", "pink", "orange", "cyan", "white")
-	new /obj/item/tool/screwdriver(src)
-	new /obj/item/tool/wirecutters(src)
-	new /obj/item/t_scanner(src)
-	new /obj/item/tool/crowbar(src)
-	new /obj/item/stack/cable_coil(src, 30, color)
-	new /obj/item/stack/cable_coil(src, 30, color)
-	if(prob(5))
-		new /obj/item/clothing/gloves/yellow(src)
-	else
-		new /obj/item/stack/cable_coil(src, 30, color)
-
+	new /obj/item/stack/cable_coil(src, 30)
+	new /obj/item/stack/cable_coil(src, 30)
```

Comment:
you coud create a full cable_coil item

Reply:
30 is a full cable coil

Reply:
you can create /obj/item/stack/cable_coil/full, rather than doing this